extra_data/tests/test_reader_mockdata.py

```diff
@@ -990,3 +991,273 @@ def test_run_metadata_no_trains(mock_scs_run):
     sel = run.select_trains(np.s_[:0])
     md = sel.run_metadata()
     assert md['dataFormatVersion'] == '1.0'
+
+
+def test_with_aliases(mock_sa3_control_data, mock_sa3_control_aliases):
+    run = H5File(mock_sa3_control_data).with_aliases(mock_sa3_control_aliases)
+
+    def assert_equal_keydata(kd1, kd2):
+        assert isinstance(kd1, KeyData)
+        assert isinstance(kd2, KeyData)
+        assert kd1.source == kd2.source
+        assert kd1.key == kd2.key
+        assert kd1.train_ids == kd2.train_ids
+
+    # Test whether source alias yields identical SourceData.
+    assert run.alias['sa3-xgm'] is run['SA3_XTD10_XGM/XGM/DOOCS']
+
+    # Test whether source alias plus literal key yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['sa3-xgm', 'pulseEnergy.wavelengthUsed'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test whether key alias yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['hv'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test undefined aliases.
+    with pytest.raises(AliasError):
+        run.alias['foo']
+        run.alias['foo', 'bar']
+
+    # Test using a literal key with a key alias.
+    with pytest.raises(ValueError):
+        run.alias['hv', 'pressure']
+
+    # Test using an existing source alias for a non-existing source.
+    with pytest.raises(SourceNameError):
+        run.alias['bogus-source']
+
+    # Test using an existing key alias for a non-existing key.
+    with pytest.raises(PropertyNameError):
+        run.alias['bogus-key']
+
+
+def test_json_alias_file(mock_sa3_control_data, mock_sa3_control_aliases):
+    with TemporaryDirectory() as td:
+        aliases_path = Path(td) / 'aliases.json'
+        aliases_path.write_text('''
```

Comment:
There's a fixture for that.

```suggestion
def test_json_alias_file(mock_sa3_control_data, mock_sa3_control_aliases, tmp_path):
    aliases_path = tmp_path / 'aliases.json'
    aliases_path.write_text('''
```